### PR TITLE
[jwt-cpp] Update jwt-cpp to 0.6.0

### PR DESCRIPTION
**Pull request by macedot**
**Describe the pull request**

- #### What does your PR fix?
Update do current stable release of jwt-cpp (Support for OpenSSL 3.0.0, Implemented the "alg" ES256K, etc)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
 All / same as before

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  `Yes`

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by Adela0814** [comment]
@macedot 

```
versions/j-/jwt-cpp.json(2,): error : While reading versions for port jwt-cpp from file: version/j-/jwt-cpp.json
Version `0.6.0#1` was not found in versions file.
Run:

    vcpkg x-add-version jwt-cpp

to add the new port version
```

**Comment by macedot** [comment]
New version added to vcpkg metadata

File: ports/jwt-cpp/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "jwt-cpp",
-  "version-semver": "0.5.1",
+  "version-semver": "0.6.0",
   "port-version": 1,
```
Comment (Adela0814) [inline ports/jwt-cpp/vcpkg.json:4]: `port-version` should be deleted after version upgrade. And please add `licenses` field for `jwt-cpp`.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Review by Adela0814 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the update!

**Pull request merged**